ring/ring.go

```diff
@@ -286,14 +286,13 @@ func (r *Ring) starting(ctx context.Context) error {
 func (r *Ring) loop(ctx context.Context) error {
 	// Update the ring metrics at start of the main loop.
 	r.updateRingMetrics()
-	go func() {
-		// Start metrics update ticker to update the ring metrics.
-		ticker := time.NewTicker(10 * time.Second)
-		defer ticker.Stop()
 
+	metricChan := make(chan struct{}, 1)
+	go func() {
 		for {
 			select {
-			case <-ticker.C:
+			case <-metricChan:
+				time.After(time.Second) // debounce and avoid consuming too much CPU
```

Comment:
why does this `time.After` reduce the CPU usage (based on comment)?

Reply:
Do I understand it correctly that the purpose of this is to prevent that `.updateRingMetrics()` gets called many times if there are many ring updates within a short time frame?

Reply:
updateRingMetrics won't be called many times, since it's much slower than KV.Watch / updateRingState, so the metricChan buffer will sink events. On the other hand if there's a burst of activity it could be worth waiting a little bit (debounce) before calculating (although it doesn't effect our tests if I leave out debounce). And in case there's sustained events long term, then we'd get updateRingState executing continuously.

I must admit it's a bit overengineered and also delays cancelling the goroutine so I can remove it if you prefer?

Reply:
Thanks for explaining.

> I can remove it if you prefer

I think it's fine to keep it, but maybe the comment could explain it a bit more because at least for me it wasn't obvious what "debounce" means in this context.

Reply:
I agree on improving the comment.

Reply:
went with different solution